docs/construct.md

```diff
@@ -0,0 +1,237 @@
+each in the interval $[0, 1)$.
+
+```python
+
+   s = sparse.random((10, 10), density=0.1)
+```
+
+Building [COO][sparse.COO] Arrays from [DOK][sparse.DOK] Arrays
+
+It's possible to build [COO][sparse.COO] arrays from [DOK][sparse.DOK] arrays, if it is not
+easy to construct the `coords` and `data` in a simple way. [DOK][sparse.DOK]
+arrays provide a simple builder interface to build [COO][sparse.COO] arrays, but at
+this time, they can do little else.
+
+You can get started by defining the shape (and optionally, datatype) of the
+[DOK][sparse.DOK] array. If you do not specify a dtype, it is inferred from the value
+dictionary or is set to `dtype('float64')` if that is not present.
+
+```python
+
+   s = DOK((6, 5, 2))
+   s2 = DOK((2, 3, 4), dtype=np.uint8)
+```
+
+After this, you can build the array by assigning arrays or scalars to elements
+or slices of the original array. Broadcasting rules are followed.
+
+```python
+
+   s[1:3, 3:1:-1] = [[6, 5]]
+```
+
+DOK arrays also support fancy indexing assignment if and only if all dimensions are indexed.
+
+```python
+
+   s[[0, 2], [2, 1], [0, 1]] = 5
+   s[[0, 3], [0, 4], [0, 1]] = [1, 5]
+```
+
+Alongside indexing assignment and retrieval, [DOK][sparse.DOK] arrays support any arbitrary broadcasting function
+to any number of arguments where the arguments can be [SparseArray][sparse.SparseArray] objects, [scipy.sparse.spmatrix][]
+objects, or [numpy.ndarray][].
+
+```python
+
+   x = sparse.random((10, 10), 0.5, format="dok")
+   y = sparse.random((10, 10), 0.5, format="dok")
+   sparse.elemwise(np.add, x, y)
+```
+
+[DOK][sparse.DOK] arrays also support standard ufuncs and operators, including comparison operators,
+in combination with other objects implementing the *numpy* *ndarray.\__array_ufunc\__* method. For example,
+the following code will perform elementwise equality comparison on the two arrays
+and return a new boolean [DOK][sparse.DOK] array.
+
+```python
+
+   x = sparse.random((10, 10), 0.5, format="dok")
+   y = np.random.random((10, 10))
+   x == y
+```
+
+[DOK][sparse.DOK] arrays are returned from elemwise functions and standard ufuncs if and only if all
+[SparseArray][sparse.SparseArray] objects are [DOK][sparse.DOK] arrays. Otherwise, a [COO][sparse.COO] array or dense array are returned.
+
+At the end, you can convert the [DOK][sparse.DOK] array to a [COO][sparse.COO] arrays.
+
+```python
+
+   s3 = COO(s)
+```
+
+In addition, it is possible to access single elements and slices of the [DOK][sparse.DOK] array
+using normal Numpy indexing, as well as fancy indexing if and only if all dimensions are indexed.
+Slicing and fancy indexing will always return a new DOK array.
+
+```python
+
+   s[1, 2, 1]  # 5
+   s[5, 1, 1]  # 0
+   s[[0, 3], [0, 4], [0, 1]] # <DOK: shape=(2,), dtype=float64, nnz=2, fill_value=0.0>
+```
+
+## Converting [COO][sparse.COO] objects to other Formats
+
+[COO][sparse.COO] arrays can be converted to [Numpy arrays][numpy.ndarray],
```

Comment:
For links to API, I'd prefer the fully qualified name, and perhaps in monospace font (using backticks in Markdown)

Reply:
You might be able to use regex find/replace to edit most of these.

Reply:
> For links to API, I'd prefer the fully qualified name,

Could you please give me an example of this? I did it to show the links with the same name as the original sphinx website.

Reply:
>in monospace font (using backticks in Markdown)

If I use backticks it doesn't resolve the link (it shows [COO][sparse.COO] as opposed to COO).

Reply:
I guess one of the following should work:

```
[][sparse.COO]
[`sparse.COO`][sparse.COO]
```

Reply:
> Could you please give me an example of this? I did it to show the links with the same name as the original sphinx website.

The full name starting with `sparse`.

Reply:
We agreed to push this out as a bonus task to the end.